model/task_lifecycle.go

```diff
@@ -762,17 +758,17 @@ func UpdateBuildAndVersionStatusForTask(taskId string, updates *StatusChanges) e
 	}
 
 	if finishedTasks >= len(buildTasks) {
-		buildComplete = true
+		updates.BuildComplete = true
 	}
-	if buildComplete || tasksToNotify >= len(buildTasks) {
+	if updates.BuildComplete || tasksToNotify >= len(buildTasks) {
```

Comment:
this can be combined with the previous if block.
`tasksToNotify` is no longer relevant.

Reply:
It's still relevant because it considers blocked tasks, whereas finishedTasks doesn't. For clarity (and to not change the compile task block at the bottom) I'd still like a different variable here, but I'll count just blocked instead of this tasksToNotify concept.